### PR TITLE
CODEX schema check

### LINKED ISSUE #9: Integrate validation of new CODEX metadata against JSON schema
@VasylVaskivskyi created a JSON schema for a new standardized version of the CODEX metadata, [here](https://github.com/hubmapconsortium/codex-common-acquisition-metadata/blob/master/metadata_standard/1.0/experiment_schema.json). Integrate into this repository so this validation is performed for new CODEX datasets.

### PR DESCRIPTION
**Pull request by mccalluc**
Fix #9

I have not run this. If it's on my shoulders, I would take the time to add tests to this repo... but if you have some other way of checking it, great: I think it's in your court @jswelling to say what the next step should be.

### CODE REVIEW

**Comment by mruffalo** [comment]
Thanks very much for jumping on this, @mccalluc -- and I should have mentioned that there's an [example metadata file](https://github.com/hubmapconsortium/codex-common-acquisition-metadata/blob/master/metadata_standard/1.0/experiment.json) in the same [directory](https://github.com/hubmapconsortium/codex-common-acquisition-metadata/tree/master/metadata_standard/1.0) as the schema, which I hope would be usable for unit testing.

File: src/ingest_validation_tests/codex_json_validator.py [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+class CodexJsonValidator(Validator):
+    description = "Check CODEX JSON against schema"
+    cost = 1.0
+
+    def collect_errors(self, **kwargs) -> List[str]:
+        schema_path = Path(__file__).parent / 'codex_schema.json'
+        schema = json.loads(schema_path.read_text)
+
+        rslt = []
+        for glob_expr in ['**/[Ee]xperiment.json']:
```
[the remark this answers is not in the document]
Comment (mruffalo) [reply src/ingest_validation_tests/codex_json_validator.py:19]: After talking with @VasylVaskivskyi, this needs to be a different (new) filename, and we settled on `dataset.json`. I will make a PR to `ingest-validation-tools` to update the directory schema to match.

**Review by mruffalo — CHANGES_REQUESTED** [review]

File: src/ingest_validation_tests/codex_json_validator.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (mccalluc) [reply src/ingest_validation_tests/codex_json_validator.py]: pushed update

**Review by mruffalo — APPROVED** [review]

**Review by jswelling — COMMENTED** [review]
I think this gets applied to every file matching **/dataset.json regardless of assay type- is that really the intent?  @mruffalo Do you want to restrict this to just CODEX?

**Comment by jswelling** [comment]
@mruffalo if you want to limit it to CODEX I think it's sufficient to just add:
```
    if not self.assay_type.lower() contains "codex":
        return
```
at the top of collect_errors().

**Comment by mruffalo** [comment]
Thanks @jswelling -- it is the intention to limit this to only CODEX datasets. @mccalluc, I'm happy to push some changes to implement that, if you're okay with me adding commits to your branch/PR.

**Comment by mruffalo** [comment]
I pushed a new branch [mruffalo/validate-codex-json-additions](https://github.com/hubmapconsortium/ingest-validation-tests/tree/mruffalo/validate-codex-json-additions) (just to err on the side of better PR etiquette), and I'm very happy to have those commits added to this PR if @mccalluc agrees.

**Comment by mccalluc** [comment]
@mruffalo , thanks! I've made
- #11

from your branch... but CI failed? Re-running now, in case it's a fluke.

**Comment by mccalluc** [comment]
Sorry for the noise: I think I'll close this, and then open a new PR on the same branch with only Joel as a reviewer. Going forward, I'm going to try not sending PR reminders, and we'll see how that goes.

**Pull request closed without merging**